src/Orleans.Runtime/MembershipService/MembershipTableManager.cs

```diff
@@ -468,7 +467,7 @@ private async Task<bool> CleanupMyTableEntries(MembershipTableData table)
                     // Declare older clone of me as Dead.
                     silosToDeclareDead.Add(tuple);   //return DeclareDead(entry, eTag, tableVersion);
                 }
-                else if (siloAddress.Generation > myAddress.Generation)
+                else if (siloAddress.Generation > myAddress.Generation && this.CurrentStatus != SiloStatus.Dead)
```

Comment:
we should clean the entry regardless the CurrentStatus is Dead or not, right? why add this &&

Reply:
We shouldn't do anything to membership once we're dead - instead we should leave that to the living.

NOTE this PR is not ready to be merged

Reply:
oh ! almost merged it. thanks for the reminder